firmware/Core/Src/littlefs/flash_driver.c

```diff
@@ -506,8 +509,18 @@ FLASH_error_enum_t FLASH_is_reachable(SPI_HandleTypeDef *hspi, uint8_t chip_numb
         are_bytes_correct = 0;
     }
 
+    // Extract memory capacity
+    uint32_t capacity_mib = (rx_buffer[2] * 512) / 32;  // Convert to Mebibytes (MiB)
```

Comment:
Use const keyword. Avoid unnecessary math (multiply-then-divide). Use bitshifts for power-of-2 operations. 

Ideally, include where you got this conversion math from.